engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -535,8 +534,12 @@ public void runLocal(AppFactory appConfig) throws Exception
 
   public URLClassLoader loadDependencies()
   {
-    URLClassLoader cl = URLClassLoader.newInstance(launchDependencies.toArray(new URL[launchDependencies.size()]));
+    URL[] dependencies = launchDependencies.toArray(new URL[launchDependencies.size()]);
+
+    ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
While it will be OK just to fix the functional issue in this PR and have JIRA open with optimization request it will be great if the contributor is willing to put a little bit more effort and implement the optimization. Said that I will not stop any other committer from merging the PR if it looks sufficient.